app/screens/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,241 @@
+                ...RIGHT_BUTTON,
+            }],
+        });
+    }, [theme.sidebarHeaderTextColor]);
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case LEFT_BUTTON.id: {
+                        onClose();
+                        break;
+                    }
+                    case RIGHT_BUTTON.id:
+                        onSaveEditedPost();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [postMessage]);
+
+    useDidUpdate(() => {
+        // Workaround to avoid iOS emdash autocorrect in Code Blocks
+        if (Platform.OS === 'ios') {
+            onPostSelectionChange();
+        }
+    }, [postMessage]);
+
+    const onPostSelectionChange = useCallback((curPos: number = cursorPosition) => {
+        if (Platform.OS === 'ios') {
+            setKeyboardType(switchKeyboardForCodeBlocks(postMessage, curPos));
+        }
+        setCursorPosition(curPos);
+    }, [cursorPosition, postMessage]);
+
+    const updateCanEditPostButton = useCallback((enabled) => {
+        if (rightButtonEnabled !== enabled) {
+            setRightButtonEnabled(enabled);
+            setButtons(componentId, {
+                leftButtons: [{...LEFT_BUTTON, icon: closeButtonIcon}],
+                rightButtons: [{...RIGHT_BUTTON, enabled}],
+            });
+        }
+    }, [closeButtonIcon, rightButtonEnabled, componentId]);
+
+    const onPostChangeText = useCallback((message: string) => {
+        setPostMessage(message);
+        const tooLong = message.trim().length > maxPostSize;
+
+        if (tooLong) {
+            const line = tooLong ? intl.formatMessage({id: 'mobile.message_length.message_split_left', defaultMessage: 'Message exceeds the character limit'}) : undefined;
+            const extra = tooLong ? `${message.trim().length} / ${maxPostSize}` : undefined;
+            setErrorLine(line);
+            setErrorExtra(extra);
+        }
+        updateCanEditPostButton(Boolean(message) && !tooLong);
+    }, [intl, maxPostSize, updateCanEditPostButton]);
+
+    const emitEditing = useCallback((loading) => {
+        setRightButtonEnabled(!loading);
+        setButtons(componentId, {
+            leftButtons: [{...LEFT_BUTTON, icon: closeButtonIcon}],
+            rightButtons: [{...RIGHT_BUTTON, enabled: !loading}],
+        });
+    }, [closeButtonIcon]);
+
+    const onClose = useCallback(() => {
+        Keyboard.dismiss();
+        popTopScreen();
+    }, []);
+
+    const onSaveEditedPost = useCallback(async () => {
+        setIsUpdating(true);
+        setErrorLine(undefined);
+        setErrorExtra(undefined);
+
+        emitEditing(true);
+
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
in any case you should use `// @ts-expect-error description`

Reply:
Corrected.